ocpmodels/preprocessing/atoms_to_graphs.py

```diff
@@ -0,0 +1,186 @@
+    else:
+        from tqdm import tqdm
+except NameError:
+    from tqdm import tqdm
+
+
+class AtomsToGraphs:
+    def __init__(
+        self, max_neigh=12, radius=6, dummy_distance=7, dummy_index=-1
+    ):
+        self.max_neigh = max_neigh
+        self.radius = radius
+        self.dummy_distance = dummy_distance
+        self.dummy_index = dummy_index
+        self.data_list = []
+
+    def _get_neighbors_pymatgen(self, atoms):
+        self.atoms = atoms
+        self.struct = AseAtomsAdaptor.get_structure(self.atoms)
+        # these return jagged arrays meaning certain atoms have more neighbors than others
+        _c_index, n_index, _images, n_distance = self.struct.get_neighbor_list(
+            r=self.radius
+        )
+        # find the delimiters, the number of neighbors varies
+        delim = np.where(np.diff(_c_index))[0] + 1
+        # split the neighbor distance and neighbor index based on delimiter
+        self.split_n_distances = np.split(n_distance, delim)
+        self.split_n_index = np.split(n_index, delim)
+
+    def _pad_arrays(self):
+        # c_index is the center index, the atom to find the neighbors of
+        c_index = np.arange(0, len(self.atoms), 1)
+        # pad c_index
+        self.all_c_index = np.repeat(c_index, self.max_neigh)
+        # add dummy variables to desired array length
+        self.all_n_index = np.full(
+            (len(self.atoms), self.max_neigh), self.dummy_index
+        )
+        self.all_distances = np.full(
+            (len(self.atoms), self.max_neigh), float(self.dummy_distance)
+        )
+
+        # loop over the stucture and replace dummy variables where values exist
+        for i, (n_index, distances) in enumerate(
+            zip(self.split_n_index, self.split_n_distances)
+        ):
+            if len(distances) == self.max_neigh:
+                self.all_n_index[i] = n_index
+                self.all_distances[i] = distances
+                continue
+            # padding arrays
+            elif len(distances) < self.max_neigh:
+                n_len = len(distances)
+                # potentially add if n_len == 0: print (increase radius)
+                self.all_n_index[i][:n_len] = n_index
+                self.all_distances[i][:n_len] = distances
+                continue
+            # removing extra values so the length is equal to max_neigh
+            # values are sorted by distance so only nearest neighbors are kept
+            elif len(distances) > self.max_neigh:
+                # this sorts the list min -> max and returns the indicies
+                sorted_dist_i = np.argsort(distances)
+                self.all_n_index[i] = n_index[sorted_dist_i[: self.max_neigh]]
+                self.all_distances[i] = distances[
+                    sorted_dist_i[: self.max_neigh]
+                ]
+
+    def _gen_features(self):
+        # expand distances in gaussian basis
+        gaussian_distances = expand_distance_gauss(self.all_distances)
```

Comment:
I realize this is there in `datasets/gasdb.py` as well, but going forward, I was thinking of removing this distance calculation from preprocessing altogether.. It's more of a modeling decision, and needs to be recomputed in the model `forward` pass anyway to make sure the computation graph is properly set up and gradients wrt positions computed correctly. See for e.g.:

https://github.com/Open-Catalyst-Project/baselines/blob/master/ocpmodels/models/cgcnn.py#L56
https://github.com/Open-Catalyst-Project/baselines/blob/master/ocpmodels/models/dimenet.py#L55

Having it anywhere outside of the model `forward` pass has also been a common source of bugs in the past, especially for the force models.

Thoughts?

Reply:
That makes sense to me, and I agree more of modeling decision. Just need to decided what to handle in this PR.